crates/circuit/src/dag_node.rs

```diff
@@ -119,13 +119,12 @@ pub struct DAGOpNode {
 #[pymethods]
 impl DAGOpNode {
     #[new]
-    #[pyo3(signature = (op, qargs=None, cargs=None, *, dag=None))]
+    #[pyo3(signature = (op, qargs=None, cargs=None, *))]
```

Comment:
You don't need the `*` anymore since there are no keyword only args:

```suggestion
    #[pyo3(signature = (op, qargs=None, cargs=None))]
```